### PR TITLE
verify /dev/mem is not present in QM partition

### PR DESCRIPTION
**Pull request by pengshanyu**
I want ensure that /dev/mem is not accesible inside QM partition.

Test Design (improved based on PR review):

1. Using test expression in QM partition:

    Execute the following command:

    `podman exec qm test -e /dev/mem`
    
2. Expected Result:

    /dev/mem does not exist, the command returns a non-zero value

3. Verify whether the test passed or not:

    PASS: if the command returns a non-zero value, it means /dev/mem is not present
    FAIL:  if the command returns 0, it means /dev/mem is present

### CODE REVIEW

**Comment by pengshanyu** [comment]
will improve it based on the review of the design

**Review by Yarboa — APPROVED** [review]
LGTM

**Review by Yarboa — APPROVED** [review]
LGTM

File: tests/ffi/dev_mem_inaccessible/test.sh
Comment (dougsland) [inline tests/ffi/dev_mem_inaccessible/test.sh:7]: why don't we just check if the command fail? Checking string output from commands are not safe IMHO.

**Comment by pengshanyu** [comment]
@dougsland @Yarboa thank you so much for the review.
This PR may be closed due to changes in test scope.

**Comment by pengshanyu** [comment]
Sorry for the confusion caused by [the comments](https://github.com/containers/qm/pull/495#issuecomment-2260152413)
According to the latest decision, I need to continue this work.

File: tests/ffi/dev_mem_inaccessible/test.sh
Comment (pengshanyu) [reply tests/ffi/dev_mem_inaccessible/test.sh:7]: Yes, you are right. But just checking if the command failed, I'm not sure if that's enough to tell /dev/mem doesn't exist.

File: tests/ffi/dev-mem_not_present/test.sh
Comment (dougsland) [inline tests/ffi/dev-mem_not_present/test.sh:11]: question, do you need more than $? != "0" ?

File: tests/ffi/dev-mem_not_present/test.sh
Comment (dougsland) [inline tests/ffi/dev-mem_not_present/test.sh:8]: $(podman exec..). probably should be enough..

Comment (dennisbrendel) [reply tests/ffi/dev-mem_not_present/test.sh:8]: There is no need for ls. Just use `test` or `[`. podman forwards the return value.

```
if podman exec -t qm test -f /dev/mem &>/dev/null; then
...
fi
```

Comment (dennisbrendel) [reply tests/ffi/dev-mem_not_present/test.sh:8]: Maybe rather use ` test -e` because it's a character special file

Comment (pengshanyu) [reply tests/ffi/dev-mem_not_present/test.sh:8]: @dougsland @dennisbrendel Thank you both for the review. 
I need to ask the design approver whether it would be better to use` test -e` instead of `ls` command.

File: tests/ffi/dev-mem_not_present/test.sh
Comment (pengshanyu) [reply tests/ffi/dev-mem_not_present/test.sh:11]: Yes, it is redundant, I will improve it.

**Comment by dennisbrendel** [comment]
Just a general comment: this is not a special property of `qm`, but of all containers created by podman. So it could also be considered a smoke test in a different place

File: tests/ffi/dev-mem_not_present/test.sh
Comment (pengshanyu) [reply tests/ffi/dev-mem_not_present/test.sh]: done

File: tests/ffi/dev-mem_not_present/test.sh
Comment (pengshanyu) [reply tests/ffi/dev-mem_not_present/test.sh]: Done
@dennisbrendel thank you so much for the suggestion.

File: tests/ffi/dev-mem_not_present/test.sh
Comment (dennisbrendel) [inline tests/ffi/dev-mem_not_present/test.sh:8]: Nitpicky: you don't need the output redirection, as `test` will only return something

Comment (dennisbrendel) [reply tests/ffi/dev-mem_not_present/test.sh:8]: That comment aside it looks good to me

Comment (pengshanyu) [reply tests/ffi/dev-mem_not_present/test.sh]: Sure, done.

File: tests/ffi/dev-mem_not_present/test.sh
Comment (KittyMae) [inline tests/ffi/dev-mem_not_present/test.sh:10]: It should not be present

File: tests/ffi/dev-mem_not_present/test.sh
Comment (qiyuann) [inline tests/ffi/dev-mem_not_present/test.sh:6]: Is it a convention to use dash `-` in the function name? 
I searched online, seems it's allowed in bash, but not recommended.

Comment (dennisbrendel) [reply tests/ffi/dev-mem_not_present/test.sh:6]: `-` is part of the "portable character set" [1] and thus allowed by at least POSIX [2]

[1] https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap06.html#tag_06_01
[2] https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_235

File: tests/ffi/dev-mem_not_present/test.sh
Comment (pengshanyu) [reply tests/ffi/dev-mem_not_present/test.sh]: @KittyMae Thank you for your careful review.
Done

File: tests/ffi/dev-mem_not_present/test.sh
Comment (pengshanyu) [reply tests/ffi/dev-mem_not_present/test.sh]: Thank you both for the careful review.
I was trying to distinguish /dev/mem from other words, but it does look a bit weird, so I changed hyphen to low-line.

File: tests/ffi/dev_mem_not_present/PURPOSE
Comment (dougsland) [inline tests/ffi/dev_mem_not_present/PURPOSE:1]: Please use this format in a next patch, I will unblock you for now.
 @pengshanyu  Also, your commit message container several duplicate signed-off-by (SOB) next time please remove the dup lines.

Title:
    Foobar 123 123

Description:
    Foobar 123 123

Input:
    Foobar 123 123

Expected Result:
    Foobar 123123

Jira:
    Foobar 123123

Comment (pengshanyu) [reply tests/ffi/dev_mem_not_present/PURPOSE:1]: Hi @dougsland, sorry, I didn't notice them. Sure, I'll pay attention next time. Thanks.

Comment (dougsland) [reply tests/ffi/dev_mem_not_present/PURPOSE:1]: Not your fault at all, we are starting to create it ;-)

File: tests/ffi/dev-mem_not_present/test.sh
Comment (qiyuann) [reply tests/ffi/dev-mem_not_present/test.sh]:
```
3.235 Name
In the shell command language, a word consisting **solely of underscores, digits, and alphabetics** from the portable character set. The first character of a name is not a digit.
```

Comment (dennisbrendel) [reply tests/ffi/dev-mem_not_present/test.sh]: > ```
> 3.235 Name
> In the shell command language, a word consisting 
> **solely of underscores, digits, and alphabetics** 
> from the portable character set. 
> The first character of a name is not a digit.
> ```

I did not read carefully enough. Thanks @qiyuann

**Pull request merged**
